### ISSUE TITLE
Question: is this repo still the 'official' way to rename electron?

### ISSUE THREAD

**Issue by rishson** [paulcbetts/grunt-build-atom-shell#33]
Not that I don't love this repo - (thanks @paulcbetts ) Its been really useful over the last couple of months, but given the stated need for this repo is to be able to rename the electron .exe on Windows, 
does electron 0.23.0 fix the original issue ([atom/atom-shell#713](https://github.com/atom/electron/issues/713)), given the Windows release notes for 0.23.0 state:

"__Renaming atom.exe no longer breaks native modules."

Is it possible to simply do:

npm i electron-prebuilt

and then:

* rename the .app, helper apps etc
* replace the .incs icon
* update the plist

If you no longer need to build from src, then could this repo move forward with additional tasks that could:

* do the plist changes
* replace the icon
* get your app ready for release (npm prune --production etc)
* asar your app
* codesign you app
* zip using ditto so autoupdater can be used on the resultant .zip.

Maybe these tasks belong in electron starter instead? Happy to help contribute grunt tasks if this is the correct assumption and direction.

**Comment by Rameshv** [comment]
+1

**Comment by paulcbetts** [comment]
Yep, you're mostly right, that was the main motivation for this package. That being said, this package still has its uses because it makes it super easy to run on non-released branches (either your own fork or being able to run on a specific commit of Electron). However, for most people getting started, `grunt-download-atom-shell` is sufficient.

@rishson can you send a PR to switch electron-starter to `grunt-download-atom-shell`? It should be fairly easy because I intentionally kept the parameters very similar

**Comment by maxkorp** [comment]
I definitely like the idea of having the task handle some of that other stuff. I have manual tasks for doing a lot of that, having a single task (or at least set of tasks in a package) would be awesome.

**Comment by maxkorp** [comment]
Just wanted to throw out there that I'd be down to help with that, if it's deemed the way you wanna take it (Although I can also understand wanting to keep this for building, in case people want that control over things still or if there are other building benefits (e.g. the correct name in the task manager in windows, etc).

**Comment by paulcbetts** [comment]
Also, if you're mostly using the `electron-prebuilt` package, https://github.com/paulcbetts/electron-rebuild will do the native module rebuilding bits of `grunt-build-atom-shell`. I'm working on the rest :)

**Comment by iamdriz** [comment]
@paulcbetts doesn't electron-prebuilt just run the app? It doesn't actually create any compiled files or anything. Plus doesn't electron-packager do all of this now? Thanks.

**Comment by paulcbetts** [comment]
>  doesn't electron-prebuilt just run the app? It doesn't actually create any compiled files or anything. Plus doesn't electron-packager do all of this now? Thanks.

You still need to rebuild the node native modules you're using against the version of Electron you're using, or else you'll have Problems(tm). That's what electron-rebuild does.

**Comment by matthiasg** [comment]
how can i find out which version of node is to be used ? e.g when looking at its not declaring which version of node it uses ... https://github.com/atom/electron/releases/tag/v0.35.0

**Comment by anaisbetts** [comment]
@matthiasg Just use electron-rebuild and you won't have to know that

**Comment by matthiasg** [comment]
@paulcbetts will try thanks

**Comment by maxkorp** [comment]
Can confirm, electron rebuild works well. There are a few modules around for renaming your copy of electron prebuilt, cant think of the names of any right now, but I haven't been too satisfied with any of them (theyre all much larger in scope than just that), so I may make one that _just renames electron-prebuilt_
